machine/v1beta1/types_azureprovider.go

```diff
@@ -215,6 +223,48 @@ type OSDisk struct {
 	CachingType string `json:"cachingType,omitempty"`
 }
 
+// DataDisk specifies the parameters that are used to add one or more data disks to the machine.
+type DataDisk struct {
+	// NameSuffix is the suffix to be appended to the machine name to generate the disk name.
+	// Each disk name will be in format <machineName>_<nameSuffix>.
+	// NameSuffix name must start and finish with an alphanumeric character and can only contain letters, numbers, underscores, periods or hyphens.
+	// The overall disk name must not exceed 80 chars in length.
+	// +kubebuilder:validation:Pattern:=`^[a-zA-Z0-9](?:[\w\.-]*[a-zA-Z0-9])?$`
+	// +kubebuilder:validation:MaxLength:=78
+	// +kubebuilder:validation:Required
+	NameSuffix string `json:"nameSuffix"`
+	// DiskSizeGB is the size in GB to assign to the data disk.
+	// +kubebuilder:validation:Minimum=4
+	// +kubebuilder:validation:Required
+	DiskSizeGB int32 `json:"diskSizeGB"`
+	// ManagedDisk specifies the Managed Disk parameters for the data disk.
+	// +optional
+	ManagedDisk ManagedDiskParameters `json:"managedDisk,omitempty"`
+	// Lun Specifies the logical unit number of the data disk. This value is used to identify data disks within the VM and therefore must be unique for each data disk attached to a VM.
+	// The value must be between 0 and 63.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=63
+	Lun *int32 `json:"lun,omitempty"`
+	// CachingType specifies the caching requirements.
+	// +optional
```

Comment:
what happens if not set?

Reply:
It won't be empty on the Machine struct underneath, which is then submitted to the cloud provider. This defaults it to the same default that would be chosen on the Azure GUI (Portal) if no options are specified for those. Which is `None`.

Reply:
As above, you need to indicate what the defaults are for *every* optional field and whether they are guaranteed forever or subject to change.

Reply:
Done.